Common/Models/WatchHistoricalGlucose.swift

```diff
@@ -10,48 +10,35 @@ import Foundation
 import HealthKit
 import LoopKit
 
-
 struct WatchHistoricalGlucose {
-    let samples: [NewGlucoseSample]
-
-    init(with samples: [StoredGlucoseSample]) {
-        self.samples = samples.map {
-            NewGlucoseSample(date: $0.startDate, quantity: $0.quantity, isDisplayOnly: $0.isDisplayOnly, wasUserEntered: $0.wasUserEntered, syncIdentifier: $0.syncIdentifier, syncVersion: 0)
-        }
-    }
+    let objects: [SyncGlucoseObject]
 }
 
-
 extension WatchHistoricalGlucose: RawRepresentable {
     typealias RawValue = [String: Any]
 
-    var rawValue: RawValue {
-        return [
-            "d": samples.map { $0.date },
-            "v": samples.map { Int16($0.quantity.doubleValue(for: .milligramsPerDeciliter)) },
-            "id": samples.map { $0.syncIdentifier },
-            "do": samples.map { $0.isDisplayOnly },
-            "ue": samples.map { $0.wasUserEntered }
-        ]
-    }
-
     init?(rawValue: RawValue) {
-        guard
-            let dates = rawValue["d"] as? [Date],
-            let values = rawValue["v"] as? [Int16],
-            let syncIdentifiers = rawValue["id"] as? [String],
-            let isDisplayOnly = rawValue["do"] as? [Bool],
-            let wasUserEntered = rawValue["ue"] as? [Bool],
-            dates.count == values.count,
-            dates.count == syncIdentifiers.count,
-            dates.count == isDisplayOnly.count,
-            dates.count == wasUserEntered.count
-        else {
+        guard let rawObjects = rawValue["o"] as? Data,
```

Comment:
[nit] curious why we are continuing with these shorten keys (e.g., `o`).

Reply:
Yes, this won't make much difference so I will use `objects` instead.

Reply:
BLE *can* be really slow. Check out this thread of someone trying to get 2KB/s and struggling. https://stackoverflow.com/questions/18240978/ble-peripheral-throughput-limit

I would normally call this kind of thing silly, but I do think we should be careful with BLE data usage.

Reply:
The current implementation will be larger to synchronize glucose data. (Same with synchronizing carb data, but there are far fewer carb entries than glucose samples).

The problem I ran into is that the syncIdentifier is now optional and the previous implementation (serializing multiple arrays to eliminate most extraneous info) does not support that.

If anyone has a good way to synchonize nils using the previous implementation please let me know and I can switch to that.

Reply:
What is the issue with serializing nils in the old method?

Reply:
Optionals appear to be an unsupported type. Specifically, getting this error message:
```
2020-10-19 13:53:06.589873-0700 WatchApp Extension[41187:388625] [WCSession Extension] sendGlucoseBackfillRequestMessage error: Error Domain=WCErrorDomain Code=7014 "Payload could not be delivered." UserInfo={NSUnderlyingError=0x8393db10 {Error Domain=WCErrorDomain Code=7010 "Payload contains unsupported type." UserInfo={NSLocalizedRecoverySuggestion=Only pass valid types., NSLocalizedDescription=Payload contains unsupported type.}}, NSLocalizedDescription=Payload could not be delivered.}
```

Reply:
Or, I should say, an *array* of optional strings are unsupported.

Reply:
Okay, updated to implementation that allows optional strings in array and only increases size of transferred data to watch by ~7%. Replaced SyncGlucoseObject with StoredGlucoseSample.

Reply:
What is the size of an average context update, btw, out of curiosity?

Reply:
IIRC, 5K for ~70 samples, I think?

Reply:
I'll check again when I run it next.

Reply:
Running with the latest code, the most # of samples that will transfer is 48, but more common full refresh is 36 samples. (I believe the ~70 samples above was because I was using the CGM simulator and had inadvertently used Backfill Glucose more than once.)

Using current data, 48 samples transfers about ~7K of data. So, each sample is about 150 bytes.

Reply:
If getting 2KB/s is challenging, even with the ability to optimize the connection interval, and we have 7KB to send, and I don't know what connection interval the watch uses, we may want to look into this when we have time (probably not a priority now). I think using enum Int keys for these structures would probably be a small boost, rather than string keys, and would read better in the code.

Reply:
Added Jira ticket to come back to: https://tidepool.atlassian.net/browse/LOOP-2161